src/extensions/registries/page-menu-registry.ts

```diff
@@ -30,28 +30,16 @@ export interface PageMenuComponents {
 export class GlobalPageMenuRegistry extends BaseRegistry<PageMenuRegistration> {
   @action
   add(items: PageMenuRegistration[], ext: LensExtension) {
-    const normalizedItems = items.map(menuItem => {
-      menuItem.target = {
-        extensionId: ext.name,
-        ...(menuItem.target || {}),
-      };
-      return menuItem;
-    });
-    return super.add(normalizedItems);
+    const norm = items.map(item => ((item.target ??= {}).extensionId = ext.name, item));
+    return super.add(norm);
   }
 }
 
 export class ClusterPageMenuRegistry extends BaseRegistry<ClusterPageMenuRegistration> {
   @action
   add(items: PageMenuRegistration[], ext: LensExtension) {
-    const normalizedItems = items.map(menuItem => {
-      menuItem.target = {
-        extensionId: ext.name,
-        ...(menuItem.target || {}),
-      };
-      return menuItem;
-    });
-    return super.add(normalizedItems);
+    const norm = items.map(item => ((item.target ??= {}).extensionId = ext.name, item));
```

Comment:
Not directly related to PR title, this just changes syntax and I'm not 💯 convinced that this is more readable.

Reply:
Isin't this also (slightly) semantically different? In the new code extensionId is always overwritten by ext.name, but in the old code extensionId would only be added if missing from target. (No idea if that has any real effect).